### PR TITLE
chore: Substrate event and deposit handlers refactor

### LINKED ISSUE #230: Refactor substrate event handlers
We want to match event handlers on EVM with event handlers on Substrate.

## Implementation details

Replace custom substrate code in the `sygma-relayer` with the code from `sygma-core` (most modules are now 
in the `core` package).
Refactor event handlers so they match the new format (instead of receiving events they are expected to fetch their own per handler to match EVM).

## Testing details

- unit tests

## Acceptance Criteria

- substrate modules are used from core package

### PR DESCRIPTION
**Pull request by mj52951**
Substrate listener subdirectory is now refactored so it doesn't use deprecated `chainbridge-core` but `sygma-core`.

closes #230 

As EVM event handlers were refactored in PR #243, this also closes issue: #235.

Note: e2e, unit and lint tests should fail as this is a step-by-step refactor focusing on smaller parts that should work

### CODE REVIEW

File: chains/substrate/config.go [hunk 1]
```diff
@@ -11,10 +11,15 @@ import (
 	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
 	"github.com/creasty/defaults"
 	"github.com/mitchellh/mapstructure"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
 
 	"github.com/ChainSafe/chainbridge-core/config/chain"
 )
 
+const (
+	FungibleTransfer message.MessageType = "FungibleTransfer"
```
Comment (mpetrun5) [inline chains/substrate/config.go:20]: Can we use this for both ERC20 and Fungible transfers and put it somewhere where it makes sense for both (outside network specific files)

File: chains/substrate/listener/deposit-handler.go [hunk 2]
```diff
@@ -6,14 +6,14 @@ package listener
...
-	core_types "github.com/ChainSafe/chainbridge-core/types"
+	"github.com/ChainSafe/sygma-relayer/chains"
+	"github.com/ChainSafe/sygma-relayer/chains/substrate"
 	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/rs/zerolog/log"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
 )
 
-type DepositHandlers map[message.TransferType]DepositHandlerFunc
+type DepositHandlers map[message.MessageType]DepositHandlerFunc
```
Comment (mpetrun5) [inline chains/substrate/listener/deposit-handler.go:16]: Why was this changed to `MessageType`. It seems that `TransferType` makes more sense in this context.

Comment (mpetrun5) [reply chains/substrate/listener/deposit-handler.go:16]: The deposit handler then creates the message

File: chains/substrate/listener/event-handlers.go [hunk 3]
```diff
@@ -6,27 +6,39 @@ package listener
...
 	"github.com/centrifuge/go-substrate-rpc-client/v4/registry"
 	"github.com/centrifuge/go-substrate-rpc-client/v4/registry/parser"
 	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
 	"github.com/mitchellh/mapstructure"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
 )
 
+type EventListener interface {
```
Comment (mpetrun5) [inline chains/substrate/listener/event-handlers.go:19]: EventFetcher would be more appropriate then

File: chains/substrate/listener/listener.go [hunk 4]
```diff
@@ -4,114 +4,31 @@
...
-				}
-				startBlock.Add(startBlock, l.blockInterval)
-			}
-		}
-	}()
-}
-
-func (l *SubstrateListener) fetchEvents(startBlock *big.Int, endBlock *big.Int) ([]*parser.Event, error) {
-	l.log.Debug().Msgf("Fetching substrate events for block range %s-%s", startBlock, endBlock)
+func (l *Listener) FetchEvents(startBlock *big.Int, endBlock *big.Int) ([]*parser.Event, error) {
```
Comment (mpetrun5) [inline chains/substrate/listener/listener.go:31]: I think we had a slight misunderstanding. This should still be ListenToEvents and the logic should be the same as before, but the only difference should be that `evts` is not fetched here, but fetched inside the specific handlers.

File: chains/substrate/listener/event-handlers.go (hunk 3 above)
Comment (mpetrun5) [reply chains/substrate/listener/event-handlers.go:19]: Though, this interface should be extra

**Review by mpetrun5 — CHANGES_REQUESTED** [review]
Ignore this until after the exams :smile:

File: chains/substrate/listener/deposit-handler.go (hunk 2 above)
Comment (mj52951) [reply chains/substrate/listener/deposit-handler.go:16]: `TransferType` existed in `chainbridge-core` so I've used `MessageType` from `sygma-core` because both are aliases for the built-in type `string`. But I agree, it is better naming to use `TransferType`, so I'll make a custom one in the repo?

Comment (mpetrun5) [reply chains/substrate/listener/deposit-handler.go:16]: yep

File: app/app.go [hunk 5]
```diff
@@ -43,6 +40,9 @@ import (
 	"github.com/sygmaprotocol/sygma-core/chains/evm/listener"
 	"github.com/sygmaprotocol/sygma-core/chains/evm/transactor/monitored"
 	"github.com/sygmaprotocol/sygma-core/chains/evm/transactor/transaction"
+	"github.com/sygmaprotocol/sygma-core/chains/substrate/client"
+	"github.com/sygmaprotocol/sygma-core/chains/substrate/connection"
+	core_substrate_listener "github.com/sygmaprotocol/sygma-core/chains/substrate/listener"
```
Comment (mpetrun5) [inline app/app.go:45]: I think generally everywhere else we use camel case.

**Review by mpetrun5 — APPROVED** [review]

File: app/app.go (hunk 5 above)
Comment (mj52951) [reply app/app.go:45]: Yes, I've made it part of the next PR already :)
